airflow/contrib/operators/sagemaker_base_operator.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+
+from airflow.contrib.hooks.sagemaker_hook import SageMakerHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class SageMakerBaseOperator(BaseOperator):
+    """
+    This is the base operator for all SageMaker operators.
+
+    :param config: The configuration necessary to start a training job (templated)
+    :type config: dict
+    :param aws_conn_id: The AWS connection ID to use.
+    :type aws_conn_id: str
+    """
+
+    template_fields = ['config']
+    template_ext = ()
+    ui_color = '#ededed'
+
+    @apply_defaults
+    def __init__(self,
+                 config,
+                 aws_conn_id='aws_default',
+                 *args, **kwargs):
+        super(SageMakerBaseOperator, self).__init__(*args, **kwargs)
+
+        self.aws_conn_id = aws_conn_id
+        self.config = config
+        self.hook = SageMakerHook(aws_conn_id=self.aws_conn_id)
```

Comment:
Hook shouldn't be created in Operator constructor

Reply:
Updated. Delayed creation to execute()